src/library/tactic/tactic_log.cpp

```diff
@@ -70,34 +81,23 @@ static tactic_log::goal summarize(metavar_decl const & d) {
 static tactic_log::summary summarize(tactic_state const & s) {
     std::vector<tactic_log::goal> goals;
     for (auto& g : s.goals()) goals.push_back(summarize(s.mctx().get_metavar_decl(g)));
-    return {s.decl_name(), std::move(goals)};
-}
-
-tactic_state_id tactic_log::get_id(tactic_state const & ts) const {
-    auto s = summarize(ts);
-    lock_guard<mutex> l(m_mutex);
-    auto& map = get_state_map(l);
-    auto it = map.find(s);
-    if (it != map.end()) return it->second;
-    tactic_state_id r = get_states(l).size();
-    get_states(l).push_back(s);
-    map.emplace(s, r);
 
+    optional<std::string> pp;
     if (get_global_module_mgr()->get_export_tspp()) {
         std::ostringstream os;
-        os << ts.pp();
-        get_tspps(l).push_back(os.str());
+        os << s.pp();
+        pp = optional<std::string>(os.str());
```

Comment:
No, this should not be done in `summarize` because that is before deduplication. It should either be like I wrote it, separate from the `summary` so that `summarize` does not produce it, or else `summarize` should always produce `pp = none`, the hash and eq should ignore this field, and it should only be set in `get_id` when we actually want to push it to `get_states`.